src/prog_algs/uncertain_data/uncertain_data.py

```diff
@@ -60,6 +60,19 @@ def cov(self) -> array:
             array[array[float]]: Covariance matrix
         """
 
+    def relative_accuracy(self, ground_truth : dict) -> dict:
+        """The relative accuracy measure of error in RUL prediction relative to the actual RUL at a specific time index
+        
+        Returns:
+            dict(str:float): Relative accuracy for each event where value is relative accuracy between [0,1]
+        """
+        # if this check isn't here, goes to divide by zero check and raises AttributeError instead of TypeError. Keep? There are unittests checking for type
+        if not isinstance(ground_truth, dict):
+            raise TypeError("Ground truth must be passed as a dictionary argument.")
```

Comment:
Ground truth could also be a *.container. We should probably check that out too. Maybe we should make UserDict a base for *.container, or manually check for it.